crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -39,9 +39,13 @@ impl FormatNodeRule<MatchCase> for FormatMatchCase {
                         write!(f, [text("case"), space()])?;
 
                         if is_match_case_pattern_parenthesized(item, pattern, f.context())? {
-                            parenthesized("(", &pattern.format(), ")")
-                                .with_dangling_comments(open_parenthesis_comments)
-                                .fmt(f)?;
+                            parenthesized(
+                                "(",
+                                &pattern.format().with_options(Parentheses::Never),
+                                ")",
+                            )
```

Comment:
Do we actually need to preserve the parentheses? Black does but in other nodes we don't:
```python
while (True):
	pass
```

Gets formatted to:
```python
while True:
	pass
```

Reply:
Our rule for expressions are:

* Top level expression (inside a statement): Remove unnecessary parentheses, see your `while (True)` example
* nested expressions: Preserve the parentheses (except for `await`  where Black removes parentheses). 

IMO the behavior should be the same for `Patterns`. Remove unnecessary parentheses around the outermost pattern but preserve them for nested patterns. See https://github.com/astral-sh/ruff/issues/6753

Meaning, we should use `parenthesize_if_expands` here instead of preserving the parentheses.

Reply:
I'll try out `parenthesize_if_expands`.

Reply:
I think this also requires knowing whether the pattern has its own parentheses (e.g., for sequence patterns).